spec/features/root_routing_spec.rb

```diff
@@ -0,0 +1,26 @@
+#
+# A routing test implemented as a feature. This is necessary so we can fake
+# logins.
+# See: https://github.com/plataformatec/devise/issues/1670#issuecomment-4060164
+#
+
+RSpec.describe 'GET /', type: :feature do
+  context 'when not logged in' do
+    it 'routes to the root_path' do
+      visit '/'
+      expect(current_path).to eq(root_path)
+    end
+  end
+
```

Comment:
No it's fine. I just read the specs before the implementation and hadn't realised that you were using conditional routing.